src/systems/filecoin_blockchain/chainsync/_index.md

```diff
@@ -1,58 +1,388 @@
+  - `Block` propagation is one of the most security critical points of the whole protocol.
+  - Bandwidth usage during `Block` propagation is the biggest rate limiter for network scalability.
+  - The time it takes for a `Block` to propagate to the whole network is a critical factor in determining a secure `BlockTime`
+  - Blocks propagating through the network should take as few _sequential_ roundtrips as possible, as these roundtrips impose serious block time delays. However, interleaved roundtrips may be fine. Meaning that `block.CIDs` may be propagated on their own, without the header, then the header without the messages, then the messages.
+  - `Blocks` will propagate over a `libp2p.PubSub`. `libp2p.PubSub.Messages` will most likely arrive multiple times at a node. Therefore, using only the `block.CID` here could make this very cheap in bandwidth (more expensive in round trips)
+  - `Blocks` in a single epoch may include the same `Messages`, and duplicate transfers can be avoided
+  - `Messages` propagate through their own `MessagePubsub`, and nodes have a significant probability of already having a large fraction of the messages in a block. Since messages are the _bulk_ of the size of a `Block`, this can present great bandwidth savings.
+- **Progressive Steps of Block Propagation**
+  - **IMPORTANT NOTES**:
+      - these can be effectively pipelined. The `receiver` is in control of what to pull, and when. It is up them to decide when to trade-off RTTs for Bandwidth.
+      - If the `sender` is propagating the block at all to `receiver`, it is in their interest to provide the full content to `receiver` when asked. Otherwise the block may not get included at all.
+      - Lots of security assumptions here -- this needs to be hyper verified, in both spec and code.
+  - **Step 1. (sender) `Send SignedBlock`**:
+      - Block propagation begins with the `sender` propagating the `block.SignedBlock` - `Gossipsub.Send(b block.SignedBlock)`
+      - This is a very light object (~200 bytes), which fits into one network packet.
+      - This has the BlockCID, the MinerAddress and the Signature. This enables parties to (a) learn that a block from a particular miner is propagating, (b) validate the `MinerAddress` and `Signature`, and decide whether to invest more resources pulling the `BlockHeader`, or the `Messages`.
+      - Note that this can propagate to ther rest of the network before the next steps complete.
+  - **Step 2. (receiver) `Pull BlockHeader`**:
+      - if `receiver` **DOES NOT** already have the `BlockHeader` for `b.BlockCID`, then:
+        - `receiver` requests `BlockHeader` from `sender`:
+            - `bh := Graphsync.Pull(sender, SelectCID(b.BlockCID))`
+        - This is a light-ish object (<4KB).
+        - This has many fields that can be validated before pulling the messages. (See **Progressive Block Validation**).
+  - **Step 3. (receiver) `Pull MessageHints` (TODO: validate/decide on this)**:
```

Comment:
This makes propagation of blocks a multi-round trip process, and this call alone will make the bandwidth concerns worse than just sending {header, msgscids}. Really not sure what is being gained here.